### PR TITLE
Minor `config/conflists.c` refactoring

### PR DESCRIPTION
**Pull request by H3rnand3zzz**
- [x] I ran valgrind when using my new feature

### How to test the functionality
Try to `/omemo end`, `/omemo start` and check how it affects `accounts` config file. You can try load, unload plugins as well.

### CODE REVIEW

File: src/config/conflists.c [hunk 1]
```diff
@@ -35,54 +35,44 @@
...
+#include "log.h"
 
 #include <string.h>
 #include <glib.h>
 
 gboolean
 conf_string_list_add(GKeyFile* keyfile, const char* const group, const char* const key, const char* const item)
 {
+    if (!item) {
+        log_warning("Invalid parameter in `conf_string_list_add`: item is NULL. keyfile=%p, group=%s, key=%s", (void*)keyfile, group, key);
```
Comment (jubalh) [inline src/config/conflists.c:47]: I'm not sure `keyfile=%p` will be useful?

Comment (H3rnand3zzz) [reply src/config/conflists.c:47]: makes sense.

Comment (H3rnand3zzz) [reply src/config/conflists.c:47]: Addressed.

**Comment by H3rnand3zzz** [comment]
## Revision-1
Rebased on master and addressed the issue pointed out by @jubalh.

**Pull request merged**